xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -1159,9 +1161,22 @@ bool CPVRClients::UpdateAndInitialiseClients(bool bInitialiseAllClients /* = fal
     }
   }
 
+  // indicate that addons have now been initialised
+  {
+    CSingleLock lock(m_critSection);
+    m_bAddonsInitialised = true;
```

Comment:
using low level objects like condition is not good either. we have CEvent for this kind of jobs